crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -669,38 +692,11 @@ impl<'src> Parser<'src> {
                 Expr::IpyEscapeCommand(self.parse_ipython_escape_command_expression())
             }
             TokenKind::String | TokenKind::FStringStart => self.parse_strings(),
-            tok @ (TokenKind::Lpar | TokenKind::Lsqb | TokenKind::Lbrace) => {
-                // We need to unset the `FOR_TARGET` in the context when parsing an expression
-                // inside a parentheses, curly brace or brackets otherwise the `in` operator of a
-                // comparison expression will not be parsed in a `for` target.
-
-                // test_ok parenthesized_compare_expr_in_for
-                // for (x in y)[0] in iter: ...
-                // for (x in y).attr in iter: ...
-
-                // test_err parenthesized_compare_expr_in_for
-                // for (x in y)() in iter: ...
-                // for (x in y) in iter: ...
-                // for (x in y, z) in iter: ...
-                // for [x in y, z] in iter: ...
-                // for {x in y, z} in iter: ...
-                let current_context = self.ctx - ParserCtxFlags::FOR_TARGET;
-                let saved_context = self.set_ctx(current_context);
-
-                let expr = match tok {
-                    TokenKind::Lpar => {
-                        let parsed_expr = self.parse_parenthesized_expression();
-                        self.restore_ctx(current_context, saved_context);
-                        return parsed_expr;
-                    }
-                    TokenKind::Lsqb => self.parse_list_like_expression(),
-                    TokenKind::Lbrace => self.parse_set_or_dict_like_expression(),
-                    _ => unreachable!(),
-                };
-
-                self.restore_ctx(current_context, saved_context);
-                expr
```

Comment:
@MichaReiser we unset the `FOR_TARGET` flag here which allowed the `in` keyword to be part of a binary expression. For example, `[a, b in c, d]`